streams/src/test/java/org/apache/kafka/streams/kstream/internals/KCogroupedStreamImplTest.java

```diff
@@ -0,0 +1,403 @@
+        groupedStream = stream.groupByKey(Grouped.with(Serdes.String(), Serdes.String()));
+        cogroupedStream = groupedStream.cogroup(MockAggregator.TOSTRING_ADDER);
+    }
+
+    @Test
+    public void cogroupTest() {
+        assertNotNull(cogroupedStream);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullKGroupedStreamOnCogroup() {
+        cogroupedStream.cogroup(null, MockAggregator.TOSTRING_ADDER);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullAggregatorOnCogroup() {
+        cogroupedStream.cogroup(groupedStream, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullInitializerOnAggregate() {
+        cogroupedStream.aggregate(null, Materialized.as("store"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullMaterMaterializedOnAggregate() {
+        cogroupedStream.aggregate(STRING_INITIALIZER, (Materialized<String, String, KeyValueStore<Bytes, byte[]>>) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullMaterStoreSupplierOnAggregate() {
+        cogroupedStream.aggregate(STRING_INITIALIZER, (StoreSupplier<KeyValueStore>) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullWindowsOnWindowedAggregate() {
+        cogroupedStream.windowedBy((Windows) null);
+    }
+
+    private static final Aggregator STRING_AGGREGATOR = (Aggregator<String, String, String>) (key, value, aggregate) ->
+        aggregate + value;
+
+    private static final Initializer STRING_INITIALIZER = () -> "";
+
+    private static final Aggregator<String, String, Integer> STRSUM_AGGREGATOR = (key, value, aggregate) ->
+        aggregate + Integer.parseInt(value);
+
+    private static final Aggregator<String, Integer, Integer> SUM_AGGREGATOR = (key, value, aggregate) ->
+        aggregate + value;
+
+    private static final Initializer SUM_INITIALIZER = new Initializer() {
+        @Override
+        public Object apply() {
+            return 0;
+        }
+    };
+
+    @Test
+    public void testCogroupBassicOneTopic() {
+
+        final KStream test = builder.stream("one", stringConsumed);
+
+        final KGroupedStream groupedOne = test.groupByKey();
+
+        final KTable customers = groupedOne.cogroup(STRING_AGGREGATOR, Materialized.as("store"))
+            .aggregate(STRING_INITIALIZER, Materialized.as("store1"));
+
+        customers.toStream().to("to-one", Produced.with(Serdes.String(), Serdes.String()));
+
+        builder.stream("to-one", stringConsumed).process(processorSupplier);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> testInputTopic = driver.createInputTopic("one", new StringSerializer(), new StringSerializer());
+            testInputTopic.pipeInput("1", "A", 0);
```

Comment:
nit: `1` -> `k1` to make clear it's the key

Shy do we need to set the timestamp explicitly?

Reply:
fixed key, we don't technically need the time stamp in all tests but I don't think it hurts

Reply:
IMHO, it's distracting as if I read the test a start to reason about the timestamps. If it does not matter we should just use the auto-assigned timestamp to avoid "noise" in the test code.